### PR TITLE
Added files via upload

### PR DESCRIPTION
**Pull request by syed-ghufran-hassan**
Created skeleton code for rust

### CODE REVIEW

File: starter_templates/.rustc_info.json [hunk 1]
```diff
@@ -0,0 +1 @@
...c_fingerprint":9501504137290250209,"outputs":{"4476964694761187371":["___.exe\nlib___.rlib\n___.dll\n___.dll\n___.lib\n___.dll\nC:\\Users\\syedghufranhassan\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\ndebug_assertions\nproc_macro\ntarget_arch=\"x86_64\"\ntarget_endian=\"little\"\ntarget_env=\"msvc\"\ntarget_family=\"windows\"\ntarget_feature=\"fxsr\"\ntarget_feature=\"sse\"\ntarget_feature=\"sse2\"\ntarget_os=\"windows\"\ntarget_pointer_width=\"64\"\ntarget_vendor=\"pc\"\nwindows\n",""],"1164083562126845933":["rustc 1.36.0 (a53f9df32 2019-07-03)\nbinary: rustc\ncommit-hash: a53f9df32fbb0b5f4382caaad8f1a46f36ea887c\ncommit-date: 2019-07-03\nhost: x86_64-pc-windows-msvc\nrelease: 1.36.0\nLLVM version: 8.0\n",""],"9311443727202421799":["___.exe\nlib___.rlib\n___.dll\n___.dll\n___.lib\n___.dll\nC:\\Users\\syedghufranhassan\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\ndebug_assertions\nproc_macro\ntarget_arch=\"x86_64\"\ntarget_endian=\"little\"\ntarget_env=\"msvc\"\ntarget_family=\"windows\"\ntarget_feature=\"fxsr\"\ntarget_feature=\"sse\"\ntarget_feature=\"sse2\"\ntarget_os=\"windows\"\ntarget_pointer_width=\"64\"\ntarget_vendor=\"pc\"\nwindows\n",""]},"successes":{}}
```
Comment (rohitpaulk) [inline starter_templates/.rustc_info.json:1]: I don't think this file needs to be added to the repository, it looks like it has machine-dependent code (I see references to `\Users\syedghufranhassan`). Maybe this should go in the `.gitignore`?

File: starter_templates/main.rs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+
+//main objective is to start TCP server on port 6379 , which is the default port that Redis uses
```
Comment (rohitpaulk) [inline starter_templates/main.rs:2]: Please use comments that are similar to [the other languages](https://github.com/codecrafters-io/languages/blob/master/starter_templates/redis/python/app/main.py). There's really only one comment needed - "Uncomment this to pass the first stage".

**Comment by rohitpaulk** [comment]
@syed-ghufran-hassan I've added support for tests to run on pull requests in https://github.com/codecrafters-io/languages/commit/2407a95ab7b64fb3e9e32554e9ef8285b2e40a00. Could you rebase this on `master` and see? You should see test results sent here.

**Comment by rohitpaulk** [comment]
To test this out, here's what you'll need to do: 

1. Add a testing command to the `Makefile`, like it has been done [here](https://github.com/codecrafters-io/languages/blob/2407a95ab7b64fb3e9e32554e9ef8285b2e40a00/Makefile#L41-L47). 

2. Add a dockerfile under [this folder](https://github.com/codecrafters-io/languages/tree/master/dockerfiles/redis)

3. Add your code under the `starter_templates` folder, similar to other languages. In your case, this will be `starter_templates/redis/rust/...`

4. Link all your files in a "definition", which will go [here](https://github.com/codecrafters-io/languages/blob/2407a95ab7b64fb3e9e32554e9ef8285b2e40a00/template_compiler/definitions.rb#L70)

File: starter_templates/.rustc_info.json (hunk 1 above)
Comment (syed-ghufran-hassan) [reply starter_templates/.rustc_info.json:1]: yes sir I am shifting it to .gitignore

File: starter_templates/main.rs (hunk 2 above)
Comment (syed-ghufran-hassan) [reply starter_templates/main.rs:2]: acknowledged sir

**Comment by syed-ghufran-hassan** [comment]
Acknowledged sir. Sir one more question that I need to ask about what
is  spawn_redis_server.sh?Actually
it was a question from my end. Regards,
Ghufran

**Comment by rohitpaulk** [comment]
@syed-ghufran-hassan `spawn_redis_server.sh` is a bash script that executes the code a user writes. You can take a look at an example here: https://github.com/codecrafters-io/redis-starter-python/blob/master/spawn_redis_server.sh

**Comment by rohitpaulk** [comment]
@syed-ghufran-hassan I've updated your branch with master, you should now see test results here 🙂

You can run the same tests locally too - just check the commands in `Makefile`.

**Comment by syed-ghufran-hassan** [comment]
Greetings sir,

Sir I followed commands from the makefile I was getting some errors among
them were

[image: image.png]

Second there was error that
[image: image.png]

So shall I install linux as I am doing it in windows.

**Comment by rohitpaulk** [comment]
@syed-ghufran-hassan could you paste your reply in GitHub and not via email? I don't think that images attached in an email appear here...

**Comment by syed-ghufran-hassan** [comment]
> @syed-ghufran-hassan could you paste your reply in GitHub and not via email? I don't think that images attached in an email appear here...

Yes Sir actually these are the two images in which it was actually showing error

image (1)
image

Sir also have you received my remaining pull requests? Were they good sir or shall I improve?

**Comment by rohitpaulk** [comment]
@syed-ghufran-hassan the error that says `bundle: command not found` points to the fact that you don't have [Bundler](https://bundler.io/) installed 🙂You'll need to install this to be able to run tests (sorry, we don't have a good contributing guide yet!)

Regarding the other error, try doing `make test_redis_rust`, that's how you run a Make command. More on this here: https://opensource.com/article/18/8/what-how-makefile

**Comment by rohitpaulk** [comment]
> Sir also have you received my remaining pull requests? Were they good sir or shall I improve?

I don't see other pull requests, just this one (https://github.com/codecrafters-io/languages/pull/9).. Which ones are you referring to?

**Comment by rohitpaulk** [comment]
@syed-ghufran-hassan Rust support for the Git challenge was recently added, you could take a look at these files: 

- https://github.com/codecrafters-io/languages/tree/master/starter_templates/git/rust
- https://github.com/codecrafters-io/languages/blob/master/dockerfiles/git/rust-1.43.Dockerfile

**Comment by syed-ghufran-hassan** [comment]
> @syed-ghufran-hassan Rust support for the Git challenge was recently added, you could take a look at these files:
> 
> * https://github.com/codecrafters-io/languages/tree/master/starter_templates/git/rust
> * https://github.com/codecrafters-io/languages/blob/master/dockerfiles/git/rust-1.43.Dockerfile

Yes sir it gave me a good idea to create the file appropriately in rust.

**Comment by syed-ghufran-hassan** [comment]
> @syed-ghufran-hassan the error that says `bundle: command not found` points to the fact that you don't have [Bundler](https://bundler.io/) installed 🙂You'll need to install this to be able to run tests (sorry, we don't have a good contributing guide yet!)
> 
> Regarding the other error, try doing `make test_redis_rust`, that's how you run a Make command. More on this here: https://opensource.com/article/18/8/what-how-makefile

Thank you sir I am now looking into it.

**Comment by syed-ghufran-hassan** [comment]
Sir these are the updates which I have done so far:

Dockerfile: 
https://github.com/syed-ghufran-hassan/languages/pull/8#issue-422716310

Makefile:
https://github.com/syed-ghufran-hassan/languages/pull/7#issue-422304145

Definitions:
https://github.com/syed-ghufran-hassan/languages/pull/6#issue-422303707

Main file:
https://github.com/syed-ghufran-hassan/languages/pull/4#issue-422303110

**Comment by syed-ghufran-hassan** [comment]
Sir one more question that after testing shall I proceed with first
challenge or shall I commit first and check if I pass the first challenge
by the following command?

git commit --allow-empty -m "Running tests"
git push origin master

Regards,

Ghufran

**Comment by rohitpaulk** [comment]
@syed-ghufran-hassan could you create all pull requests against this repository please? The ones you linked are against your fork (`syed-ghufran-hassan/languages`). When you create pull requests against this repository,  you'll see tests automatically run in GitHub 🙂 Let me know if you run into any issues!

**Comment by rohitpaulk** [comment]
And it's okay to just have one pull request and not multiple, don't worry about splitting them per file.

**Comment by syed-ghufran-hassan** [comment]
> @syed-ghufran-hassan could you create all pull requests against this repository please? The ones you linked are against your fork (`syed-ghufran-hassan/languages`). When you create pull requests against this repository, you'll see tests automatically run in GitHub 🙂 Let me know if you run into any issues!

Sure sir I am doing it I will ask you if I stuck at any point

**Comment by syed-ghufran-hassan** [comment]
thank you sir

**Comment by rohitpaulk** [comment]
Closing this in favour of https://github.com/codecrafters-io/languages/pull/21

**Pull request closed without merging**